baseapp/abci.go

```diff
@@ -266,7 +278,10 @@ func (app *BaseApp) PrepareProposal(req abci.RequestPrepareProposal) abci.Respon
 // Ref: https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-060-abci-1.0.md
 // Ref: https://github.com/tendermint/tendermint/blob/main/spec/abci/abci%2B%2B_basic_concepts.md
 func (app *BaseApp) ProcessProposal(req abci.RequestProcessProposal) abci.ResponseProcessProposal {
-	// TODO: Implement.
+	err := app.processProposal(req)
+	if err != nil {
+		return abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_REJECT}
+	}
 	return abci.ResponseProcessProposal{Status: abci.ResponseProcessProposal_ACCEPT}
```

Comment:
The application is entirely responsible for `ProcessProposal` -- we shouldn't be returning any status here. `ProcessProposal` should look something like:

```go
func (app *BaseApp) ProcessProposal(req abci.RequestProcessProposal) abci.ResponseProcessProposal {
  // Update fields on the context that will be useful for applications when processing a
  // RequestProcessProposal.
  app.processProposalState.ctx = app.processProposalState.ctx.WithFoo(...)

  return app.processProposal(req) // <- defined/set by the app
}
```